src/main/kotlin/kotlinx/kover/engines/intellij/IntellijCoverage.kt

```diff
@@ -95,8 +97,9 @@ private fun Writer.writeModuleReportJson(
     appendLine("""  "modules": [""")
     appendLine("""    { "reports": [ """)
 
+    val smapIterator = smapFiles.iterator()
     appendLine(binaryReportFiles.joinToString(",\n        ", "        ") { f ->
-        """{"ic": "${f.safePath()}", "smap": "${f.safePath()}.smap"}"""
+        """{"ic": "${f.safePath()}", "smap": "${smapIterator.next().safePath()}"}"""
```

Comment:
You can remove `smapIterator` and just use `smapIterator.first()`

Reply:
Unfortunately, `first` will not be enough - it is necessary to read all the elements of the iterator inside a loop strictly in order

Reply:
Oh, I see, it's captured in lambda.